### PR TITLE
Feature/update phpunit 8 to 9

### PR DESCRIPTION
**Pull request by Chris53897**
All Changes are related to php 7.4 and phpunit 9 and to get tests running.
This PR is related to https://github.com/Payum/PayumBundle/pull/535

Serveral changes are needed in composer.json
"league/uri-schemes": "^1.1",  => Otherwise it would not install on php 8.x
"payum/payum": "^1.6.2", => Otherwise --prefer-lowest Test will not be an installable set
"symfony/phpunit-bridge": "6.0", => Only one version is needed for require-dev
"psr/log": "^1 || ^2", is more a temporary fix. I guesss the real problem is in payum itself. https://github.com/Payum/Payum/pull/926
"symfony/form": "^4.4.20 || ^5.0", => Required for the test (Annotated Validations) CreditCardTypeTest->shouldRequireExpireAtInFuture()

Update: All Tests are migrated to phpUnit 9.

Symfony4 needs some extra config to work with Symfony4, 5 and 6 in the near future.
Symfony6 will need extra Work, as there are deprecated functions.
Should be handled in an different PR, after support of Symfony6 is added.

### CODE REVIEW

**Review by pierredup — COMMENTED** [review]
Thanks for the massive effort here, just left a few small comments

File: Tests/Functional/app/config/config.yml [hunk 1]
```diff
@@ -6,14 +6,12 @@ framework:
...
-    session:
-        storage_id: 'session.storage.mock_file'
     secret:          '%secret%'
-    router:          { resource: '%kernel.project_dir%/Tests/Functional/app/config/routing.yml' }
+    router:          { utf8: true, resource: '%kernel.project_dir%/Tests/Functional/app/config/routing.yml' }
     default_locale:  '%locale%'
     form:                                                 true
     csrf_protection:                                      true
-    validation:                                           { enable_annotations: true }
+    #validation:                                           { enable_annotations: false }
```
[the remark this answers is not in the document]
Comment (Chris53897) [reply Tests/Functional/app/config/config.yml:14]: I am not sure if this config-line needs to be different between symfony 4 and 5. Or my latest changes to composer.json will fix this. But of course the commented line could be removed.

File: Tests/Functional/PayumBuilderTest.php [hunk 2]
```diff
@@ -9,58 +9,73 @@
...
-    public function testShouldContainCoreGatewayFactoryBuilder()
+
+    public function testShouldContainCoreGatewayFactoryBuilder(): void
     {
         /** @var PayumBuilder $builder */
         $builder = static::$container->get('payum.builder');
 
-        $this->assertAttributeInstanceOf(CoreGatewayFactoryBuilder::class, 'coreGatewayFactory', $builder);
+        $reflected_constraint = (new \ReflectionObject($builder))->getProperty('coreGatewayFactory');
+        $reflected_constraint->setAccessible(TRUE);
```
[the remark this answers is not in the document]
Comment (Chris53897) [reply Tests/Functional/PayumBuilderTest.php:27]: Ok, i wil replace all with my next commit

File: Tests/Functional/app/config/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Chris53897) [reply Tests/Functional/app/config/config.yml:14]: I removed the line in next commit

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -43,8 +42,8 @@ jobs:
       - name: Install PHP 8 dependencies
         if: startsWith(matrix.php, '8')
         run: |
-            composer req --dev --no-update payum/payum:dev-master
-            composer rem --dev --no-update league/uri-schemes
+          composer req --dev --no-update payum/payum:dev-master
+          composer rem --dev --no-update league/uri-schemes
```
[the remark this answers is not in the document]
Comment (Chris53897) [reply .github/workflows/tests.yaml:46]: Thanks for the background-info. I removed this step.
I guess the two Jobs can be mergedd two one with variables.
But this should be done in an different PR in my point of view.

File: Tests/Controller/AbstractControllerTest.php [hunk 4]
```diff
@@ -16,12 +17,12 @@ abstract class AbstractControllerTest extends TestCase
...
-    protected $token;
-    protected $httpRequestVerifierMock;
-    protected $gatewayMock;
-    protected $registryMock;
-    protected $payum;
-    protected $request;
+    protected Token $token;
+    protected HttpRequestVerifierInterface $httpRequestVerifierMock;
+    protected GatewayInterface $gatewayMock;
+    protected RegistryInterface $registryMock;
```
[the remark this answers is not in the document]
Comment (Chris53897) [reply Tests/Controller/AbstractControllerTest.php:23]: I changed the 3 MockObjects and let the other 3 with concrete types (no mock objects). 
Is that ok for you?

**Comment by Chris53897** [comment]
Thanks for the fast response.
I made the suggested changes. Can you please review again.

**Review by pierredup — APPROVED** [review]
Thanks for the massive effort here @Chris53897

**Pull request merged**